salt/grains/core.py

```diff
@@ -2515,7 +2515,7 @@ def _linux_iqn():
     if os.path.isfile(initiator):
         with salt.utils.files.fopen(initiator, 'r') as _iscsi:
             for line in _iscsi:
-                if line.find('InitiatorName') != -1:
+                if line.find('InitiatorName=') != -1:
```

Comment:
This will also match:
```#InitiatorName=iqn.1993-08.org.debian:01:8b806bf3f071```
or
```ErrorInitiatorName=iqn.1993-08.org.debian:01:8b806bf3f071```

Maybe use we should use `if line.find('InitiatorName=') == 0:`

Reply:
@keesbos Well if a user is going to mess with the standard, do not touch, file then I guess that could be classed as 'buyer beware'. Those scenarios you suggest can never happen as the initiatorname file is autogenerated by the iSCSI package and if we are deploying these packages using Salt then the file isn't going to be modified.

Reply:
There will be use cases and scenarios I haven't thought of.

The existence of the file `/etc/iscsi/initiatorname.iscsi` doesn't mean it's used and/or validates. On my ubuntu test container I don't use iscsi, but the file exists (and package is installed, probably by default).

The use of salt on a system doesn't necessarily imply that it is also used to configure the system (salt can be very useful for other stuff than configuration). I've seen systems with salt that where manually configured, or via puppet or ansible. Even if salt is used to configure the system, it doesn't have to be used to do every bit.

Someday, a distro might decide to generate a (valid) file that contains
```
## blah blah InitiatorName=<value> blah blah
```

Finally, what's the problem with `== 0` in stead of `!= -1` ?

Reply:
The file is there but is it populated with information?  As @sdodsley said, it's a generated file that shouldn't be edited by hand.  That being said, perhaps the `line.find` should be swapped out for a `line.startswith` similar to the code in the Windows function, since the correct line should always start with the string `InitiatorName=`.

Reply:
Yep. That would be great (`line.startswith`)